lib/miq_expression/field.rb

```diff
@@ -82,6 +82,19 @@ def contains(other)
     )
   end
 
+  def custom_attribute_column_name
+    column.gsub(CustomAttributeMixin::CUSTOM_ATTRIBUTES_PREFIX, "")
+  end
+
+  def column_type
+    if custom_attribute_column?
+      field_type = CustomAttribute.find_by(:name => custom_attribute_column_name).try(:serialized_value).class
+      field_type == NilClass ? :string : field_type.to_s.downcase.to_sym
```

Comment:
@imtayadeway Did some checks and appearntly NilClass won't supply the truth/falsey we expect, going back to == NilClass.